packages/dev/loaders/src/glTF/2.0/Extensions/KHR_xmp_json_ld.ts

```diff
@@ -53,8 +53,8 @@ export class KHR_xmp_json_ld implements IGLTFLoaderExtension {
         if (xmp_gltf && xmp_node) {
             const packet = +xmp_node.packet;
             if (xmp_gltf.packets && packet < xmp_gltf.packets.length) {
-                this._loader.rootBabylonMesh.metadata = this._loader.rootBabylonMesh.metadata || {};
-                this._loader.rootBabylonMesh.metadata.xmp = xmp_gltf.packets[packet];
+                this._loader.rootBabylonMesh._internalMetadata = this._loader.rootBabylonMesh._internalMetadata || {};
+                this._loader.rootBabylonMesh._internalMetadata.xmp = xmp_gltf.packets[packet];

```

Comment:
XMP data is not internal. Users should be able to access without using underscore properties. But it also can't be part of metadata if all of it will be serialized. Maybe this should be a special property? I also wonder if this is a breaking change if people are relying on the XMP data at this location.

Reply:
In this case, shouldn't there be a specific way to access this XMP data (like a static method in `KHR_xmp_json_ld` that would allow you to retrieve the XMP data from a mesh instance)? But of course, this would be a breaking change...

If the `metadata` property is only for end-user use, we should not modify it at the risk of nasty side effects. Documenting the properties of `metadata` that the user should not alter is a way around the problem, but it doesn't feel right...

Personally, I don't really like adding exceptions to the serialization of `metadata`, it might become a maintenance/back compat issue later (because we may add more internal properties to `metadata` and potentially break some existing user code).

If the outcome is to keep `metadata.xmp`, then I think this PR should be closed and replaced by another one, because introducing `_internalMetadata` would be useless since the original intention was to avoid having exceptions to the serialization of the `metadata` property .

Reply:
We discussed this offline and decided to revert the changes regarding XMP, because XMP data is user metadata, so it should be available on the `metadata` property and should be serialized with the other mesh data.

We are keeping all other changes related to `_internalMetadata` because the data is really for internal usage.